rnote-ui/src/workspacebrowser/mod.rs

```diff
@@ -163,64 +163,220 @@ impl WorkspaceBrowser {
+            adw::prelude::ActionGroupExt::activate_action(&appwindow, "edit-workspace", None);
+        }),
+    );
+}
+
+fn setup_edit_workspace_button(wb: &WorkspaceBrowser, appwindow: &RnoteAppWindow) {
+    wb.imp()
+        .edit_workspace_button
+        .get()
+        .connect_clicked(clone!(@weak appwindow => move |_| {
+            adw::prelude::ActionGroupExt::activate_action(&appwindow, "edit-workspace", None);
         }));
+}
 
-        // Setup prefix listbox
-        self.imp().files_prefix_listbox.connect_row_activated(clone!(@weak self as workspacebrowser, @weak appwindow => move |_, row| {
-            if row == &workspacebrowser.imp().dir_up_row.get() {
-                if let Some(parent_dir) = workspacebrowser.selected_workspace_dir().and_then(|p| p.parent().map(|p| p.to_path_buf())) {
-                    workspacebrowser.set_current_workspace_dir(parent_dir.to_path_buf());
-                }
+fn setup_workspacelist(wb: &WorkspaceBrowser, appwindow: &RnoteAppWindow) {
+    wb.imp().workspace_list.connect_items_changed(
+        clone!(@weak wb, @weak appwindow => move |folders_model, _, _, _| {
+            wb.imp().remove_workspace_button.get().set_sensitive(folders_model.n_items() > 1);
+            wb.imp().edit_workspace_button.get().set_sensitive(folders_model.n_items() > 0);
+
+            wb.save_to_settings(&appwindow.app_settings());
+        }),
+    );
+}
 
+fn setup_workspace_listbox(wb: &WorkspaceBrowser, appwindow: &RnoteAppWindow) {
+    let workspace_listbox = wb.imp().workspace_listbox.get();
+    workspace_listbox.connect_selected_rows_changed(clone!(@weak appwindow, @weak wb => move |_| {
+        if let Some(dir) = wb.current_selected_workspace_row().map(|row| row.entry().dir()) {
+            wb.imp().files_dirlist.set_file(Some(&gio::File::for_path(dir)));
+
+            wb.save_to_settings(&appwindow.app_settings());
+        }
+
+    }));
+
+    workspace_listbox.bind_model(
+        Some(&wb.imp().workspace_list),
+        clone!(@strong appwindow => move |obj| {
+            let entry = obj.to_owned().downcast::<WorkspaceListEntry>().unwrap();
+            let workspace_row = WorkspaceRow::new(entry);
+            workspace_row.init(&appwindow);
+
+            workspace_row.upcast::<Widget>()
+        }),
+    );
+}
+
+fn setup_prefix_listbox(wb: &WorkspaceBrowser, appwindow: &RnoteAppWindow) {
+    wb.imp().files_prefix_listbox.connect_row_activated(
+        clone!(@weak wb, @weak appwindow => move |_, row| {
+            if row == &wb.imp().dir_up_row.get() {
+                if let Some(parent_dir) = wb.selected_workspace_dir().and_then(|p| p.parent().map(|p| p.to_path_buf())) {
+                    wb.set_current_workspace_dir(parent_dir.to_path_buf());
+                }
             }
         }));
+}
 
-        // Setup file rows
-        {
-            let primary_list_factory = SignalListItemFactory::new();
+fn setup_file_rows(wb: &WorkspaceBrowser, appwindow: &RnoteAppWindow) {
```

Comment:
I'm a bit unsure about this function... it still has a lot of lines and it contains different logic steps, like setup the FileFilter, create the sorter and prepare some other widgets. I'd like to split this up, but I'm unsure how. On the one hand I'd like to move this function into another file since the extra functions would be only used for this function, but on the other hand it might be a little bit too much, to create only *one* extra file for this. What do you think? (I'm a little bit tempted to do the same as my previous PR: Creating a new directory and then the different files xD)

Reply:
I think its okay, I really don't think there is any benefit to split it into smaller parts. The setup here is interconnectect and are understood together. It reduces readability when you have to jump around across files to read it.

Also: why obsess over some relatively trivial and cosmetic code changes for something that already works?

Reply:
>  It reduces readability when you have to jump around across files to read it.

Hm... this can be normally avoided by using keybindings and I think that most editors and IDE's support this feature, but I do get your point. Splitting the code too much up, can result to unreadable code instead of more readable.

> Also: why obsess over some relatively trivial and cosmetic code changes for something that already works?

I'd say: Readability and Maintainability :D (but yeah, this is probably a little bit *too* much "makeup" for the code.